File: app/src/main/java/com/foxek/simpletimer/ui/base/BaseDialog.java
package com.foxek.simpletimer.ui.base;

import androidx.fragment.app.DialogFragment;

public abstract class BaseDialog<T extends  MvpMultiPresenter> extends DialogFragment implements MvpDialog<T> {

    private T presenter;

    @Override
    public void attachPresenter(T mvpPresenter) {
        presenter = mvpPresenter;
    }

    protected T getPresenter(){
        return presenter;
    }


}

File: app/src/main/java/com/foxek/simpletimer/ui/base/BaseDialog.kt
package com.foxek.simpletimer.ui.base

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.EditText
import androidx.fragment.app.DialogFragment
import com.foxek.simpletimer.di.component.ActivityComponent
import kotlinx.android.synthetic.main.dialog_edit_interval.*

abstract class BaseDialog : DialogFragment(), MvpView {

    abstract val layoutId: Int
    abstract val dialogTag: String

    val activityComponent: ActivityComponent?
        get() = (targetFragment as BaseFragment).component

    override fun onCreateView(inflater: LayoutInflater, parent: ViewGroup?, bundle: Bundle?): View? {
        dialog?.setCanceledOnTouchOutside(true)
        return inflater.inflate(layoutId, parent, false)
    }

    fun checkNotEmpty(vararg widgets: EditText): Boolean {
        return widgets.all { it.text.isNotEmpty() }
    }

}